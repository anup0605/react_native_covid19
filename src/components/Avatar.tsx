import * as React from 'react';
import { Image, ImageSourcePropType, StyleSheet } from 'react-native';

interface IProps {
  source: ImageSourcePropType;
}

export function Avatar(props: IProps) {
  return <Image source={props.source} style={styles.image} />;
}

const styles = StyleSheet.create({
  image: {
    borderRadius: 60,
    height: 60,
    overflow: 'hidden',
    resizeMode: 'contain',
    width: 60,
  },
});
