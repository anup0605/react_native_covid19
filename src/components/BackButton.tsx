import NavigatorService from '@covid/NavigatorService';
import { sizes } from '@covid/themes';
import { colors } from '@theme';
import { Icon } from 'native-base';
import * as React from 'react';
import { StyleProp, StyleSheet, TouchableOpacity, View, ViewStyle } from 'react-native';

type TProps = {
  style?: StyleProp<ViewStyle>;
};

export function BackButton(props: TProps) {
  return (
    <TouchableOpacity onPress={NavigatorService.goBack} style={props.style} testID="button-back-navigation">
      <View style={styles.iconButton}>
        <Icon name="chevron-thin-left" style={styles.icon} type="Entypo" />
      </View>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  icon: {
    color: colors.secondary,
    fontSize: 16,
  },
  iconButton: {
    alignItems: 'center',
    backgroundColor: colors.backgroundFour,
    borderRadius: sizes.m,
    height: 32,
    justifyContent: 'center',
    width: 32,
  },
});
