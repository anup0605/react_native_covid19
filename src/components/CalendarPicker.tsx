import { isUSCountry } from '@covid/core/localisation/LocalisationService';
import { sizes, styling } from '@covid/themes';
import { colors } from '@theme';
import * as React from 'react';
import { LayoutChangeEvent, StyleSheet, View } from 'react-native';
import RNCalendarPicker, { CalendarPickerProps } from 'react-native-calendar-picker';

const CalendarPicker = (props: CalendarPickerProps) => {
  const [width, setWidth] = React.useState(0);

  function onLayout(event: LayoutChangeEvent) {
    setWidth(event.nativeEvent.layout.width);
  }
  return (
    <View style={styles.calendarView}>
      <View onLayout={onLayout} style={styling.measureWidth} />
      <RNCalendarPicker
        {...props}
        selectedDayStyle={styles.selectedDay}
        selectedDayTextColor={colors.white}
        selectedRangeStyle={styles.selectedRange}
        startFromMonday={!isUSCountry()}
        todayBackgroundColor="transparent"
        todayTextStyle={styles.today}
        width={Math.min(width, sizes.maxScreenWidth)}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  calendarView: {
    backgroundColor: colors.backgroundTertiary,
    borderRadius: sizes.xs,
    padding: sizes.xs,
  },
  selectedDay: {
    backgroundColor: colors.lightBlueBrand,
  },
  selectedRange: {
    backgroundColor: colors.lightBlueBrand,
  },
  today: {
    color: colors.purple,
    fontFamily: 'SofiaPro-Bold',
  },
});

export default CalendarPicker;
