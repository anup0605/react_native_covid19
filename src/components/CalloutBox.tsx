import { vaccineBg } from '@assets';
import AnnouncementIcon from '@assets/icons/AnnouncementIcon';
import { TScreenContent } from '@covid/core/content/ScreenContentContracts';
import { sizes } from '@covid/themes';
import { colors } from '@theme';
import * as React from 'react';
import { Image, StyleProp, StyleSheet, TextStyle, TouchableOpacity, View, ViewStyle } from 'react-native';

import { RegularText } from './Text';

type TProps = {
  content: TScreenContent;
  onPress?: () => void;
  image?: boolean;
  boxStyle?: StyleProp<ViewStyle>;
  titleStyle?: StyleProp<TextStyle>;
  linkStyle?: StyleProp<TextStyle>;
};

export const CalloutBox = ({ content, boxStyle, titleStyle, linkStyle, onPress, image }: TProps) => {
  return (
    <TouchableOpacity onPress={onPress} style={[styles.discoveriesContainer, boxStyle]}>
      {image ? <Image source={vaccineBg} style={[styles.backgroundImage, { borderRadius: sizes.m }]} /> : null}
      <View style={[styles.discoveriesTitleBackground, titleStyle]}>
        <AnnouncementIcon />
        <RegularText style={styles.discoveriesTitle}>{content.title_text}</RegularText>
      </View>
      <RegularText style={styles.discoveriesText}>{content.body_text}</RegularText>
      <RegularText style={[styles.discoveriesVisitText, linkStyle]}>{content.link_text}</RegularText>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  backgroundImage: {
    height: '125%',
    left: 0,
    position: 'absolute',
    top: 0,
    width: '100%',
  },
  discoveriesContainer: {
    alignItems: 'center',
    borderColor: colors.backgroundSecondary,
    borderRadius: sizes.m,
    borderWidth: 1,
    justifyContent: 'space-between',
    marginBottom: sizes.xl,
    paddingVertical: sizes.m,
    width: '100%',
  },
  discoveriesText: {
    color: colors.white,
    fontSize: 16,
    lineHeight: 24,
    marginHorizontal: sizes.xxl,
    marginVertical: sizes.m,
    textAlign: 'center',
  },
  discoveriesTitle: {
    color: colors.white,
    fontSize: 12,
    letterSpacing: 1,
    paddingHorizontal: sizes.xxs,
  },
  discoveriesTitleBackground: {
    alignItems: 'center',
    backgroundColor: colors.lightBlueBrand,
    borderRadius: sizes.xxs,
    flexDirection: 'row',
    paddingHorizontal: sizes.xxs,
  },
  discoveriesVisitText: {
    color: colors.lightBrand,
    fontSize: 16,
    lineHeight: 24,
  },
});
