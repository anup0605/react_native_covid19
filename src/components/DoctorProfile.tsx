import Text from '@covid/components/typography/text';
import { sizes } from '@covid/themes';
import * as React from 'react';
import { StyleSheet, View } from 'react-native';

interface IProps {
  imageNode: React.ReactNode;
  location: string;
  name: string;
  title: string;
}

export default function DoctorProfile(props: IProps) {
  return (
    <View style={styles.wrapper}>
      {props.imageNode}
      <View style={styles.view}>
        <Text>{props.name}</Text>
        <Text style={styles.text} textClass="pSmall">
          {props.title}
        </Text>
        <Text style={styles.text} textClass="pSmall">
          {props.location}
        </Text>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  text: {
    color: '#888B8C',
  },
  view: {
    marginLeft: sizes.m,
  },
  wrapper: {
    alignItems: 'center',
    flexDirection: 'row',
    marginBottom: sizes.m,
  },
});
