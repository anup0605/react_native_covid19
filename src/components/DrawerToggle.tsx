import { menuIcon } from '@assets';
import { TScreenName } from '@covid/core/Coordinator';
import { TScreenParamList } from '@covid/features/ScreenParamList';
import { sizes } from '@covid/themes';
import { DrawerNavigationProp } from '@react-navigation/drawer';
import { colors } from '@theme';
import * as React from 'react';
import { Image, ImageStyle, StyleProp, StyleSheet, TouchableOpacity } from 'react-native';

type TProps = {
  navigation: DrawerNavigationProp<TScreenParamList, TScreenName>;
  style?: StyleProp<ImageStyle>;
  testID?: string;
};

export const DrawerToggle: React.FC<TProps> = (props) => (
  <TouchableOpacity
    accessible
    accessibilityLabel="Menu"
    accessibilityRole="button"
    onPress={() => {
      props.navigation.toggleDrawer();
    }}
    testID={props.testID}
  >
    <Image source={menuIcon} style={[styles.menuIcon, props.style]} />
  </TouchableOpacity>
);

const styles = StyleSheet.create({
  indicator: {
    position: 'absolute',
    right: 0,
  },
  menuIcon: {
    alignSelf: 'flex-end',
    height: sizes.drawerToggle,
    marginRight: sizes.m,
    tintColor: colors.primary,
    width: sizes.drawerToggle,
  },
});
