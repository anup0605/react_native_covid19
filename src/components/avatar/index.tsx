import * as React from 'react';
import { Image, ImageSourcePropType, StyleSheet, View } from 'react-native';

interface IProps {
  source: ImageSourcePropType;
}

export function Avatar(props: IProps) {
  return (
    <View style={styles.view}>
      <Image source={props.source} style={styles.image} />
    </View>
  );
}

const styles = StyleSheet.create({
  image: {
    aspectRatio: 1.0,
    height: undefined,
    resizeMode: 'contain',
    width: 60,
  },
  view: {
    alignItems: 'center',
    backgroundColor: 'pink',
    borderColor: 'white',
    borderRadius: 60,
    borderWidth: 2,
    height: 60,
    justifyContent: 'center',
    overflow: 'hidden',
    width: 60,
  },
});
