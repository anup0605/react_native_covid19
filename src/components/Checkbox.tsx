import Check from '@assets/icons/Check';
import { requiredFormMarker } from '@covid/components/Form';
import { sizes } from '@covid/themes';
import { colors } from '@theme/colors';
import { Item } from 'native-base';
import * as React from 'react';
import { StyleProp, StyleSheet, TouchableOpacity, View, ViewStyle } from 'react-native';

import { RegularText } from './Text';
import { ITest } from './types';

interface ICheckboxProps extends ITest {
  value: boolean;
  onChange: (value: boolean) => void;
  children: React.ReactNode;
  dark?: boolean;
  style?: StyleProp<ViewStyle>;
  testID?: string;
}

interface ICheckboxListProps {
  children: React.ReactNode;
  label?: string;
  required?: boolean;
}

export function CheckboxItem(props: ICheckboxProps) {
  return (
    <Item style={[styles.checkboxRow, props.style]}>
      <TouchableOpacity
        accessible
        accessibilityRole="checkbox"
        onPress={() => props.onChange(!props.value)}
        style={props.dark ? styles.checkBoxDark : styles.checkBox}
        testID={props.testID}
      >
        {props.value ? <Check /> : null}
      </TouchableOpacity>
      <Item onPress={() => props.onChange(!props.value)} style={styles.checkBoxText}>
        <RegularText style={{ ...styles.checkboxLabel }}>{props.children}</RegularText>
      </Item>
    </Item>
  );
}

export function CheckboxList({ children, label, required }: ICheckboxListProps) {
  const renderLabel = () =>
    required ? (
      <RegularText>
        {label}
        {required ? `${requiredFormMarker}` : null}
      </RegularText>
    ) : null;

  return (
    <View style={styles.checkboxList}>
      {renderLabel()}
      {children}
    </View>
  );
}

const checkBoxStyle: ViewStyle = {
  alignItems: 'center',
  backgroundColor: colors.backgroundTertiary,
  borderColor: 'transparent',
  borderRadius: sizes.xs,
  borderWidth: 1,
  display: 'flex',
  height: 32,
  justifyContent: 'space-evenly',
  width: 32,
};

const styles = StyleSheet.create({
  checkBox: checkBoxStyle,
  checkBoxDark: {
    ...checkBoxStyle,
    borderColor: '#C4C4C4',
  },
  checkBoxText: {
    borderColor: 'transparent',
  },
  checkboxLabel: {
    marginLeft: sizes.m,
    marginRight: sizes.xl,
    paddingTop: sizes.xxs,
  },
  checkboxList: {
    width: '100%',
  },
  checkboxRow: {
    alignItems: 'flex-start',
    borderColor: 'transparent',
    flexDirection: 'row',
    paddingVertical: sizes.xs,
  },
});
