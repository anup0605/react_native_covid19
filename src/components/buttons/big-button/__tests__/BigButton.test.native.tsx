/* eslint-env jest */

import { BigButton } from '@covid/components';
import * as React from 'react';
import renderer from 'react-test-renderer';

describe('Button', () => {
  it('should match snapshot', () => {
    const component = renderer.create(<BigButton onPress={() => {}}>Button Text</BigButton>);
    // @ts-expect-error
    expect(component.toJSON()).toMatchSnapshot();
  });
});
