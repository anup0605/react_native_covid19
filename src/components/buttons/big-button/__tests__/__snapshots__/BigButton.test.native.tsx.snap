// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="button-test-ID"
>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#565A5C",
        "borderRadius": 24,
        "borderWidth": 1,
        "fontSize": 16,
        "paddingHorizontal": 24,
        "paddingVertical": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#565A5C",
          "textAlign": "center",
        }
      }
    >
      Button Text
    </Text>
  </View>
</View>
`;
